torch/csrc/lazy/core/shape_inference.cpp

```diff
@@ -418,6 +420,52 @@ std::vector<Shape> compute_shape_embedding_dense_backward(
       Shape(grad_output.scalar_type(), {num_weights, grad_output.size(-1)})};
 }
 
+std::vector<Shape> compute_shape_expand(
+    const at::Tensor& self,
+    at::IntArrayRef size,
+    bool implicit) {
+  CHECK_GE(size.size(), self.dim());
+  int64_t num_new_dimensions = size.size() - self.dim();
+  std::vector<int64_t> padded_self(num_new_dimensions, 0);
+  padded_self.insert(
+      padded_self.end(), self.sizes().begin(), self.sizes().end());
+  std::vector<int64_t> target_size(size.size());
+  for (const auto idx : c10::irange(size.size())) {
+    target_size[idx] = size[idx] == -1 ? padded_self[idx] : size[idx];
```

Comment:
@miladm are you sure this is correct? 

If I run the example below, PyTorch produces an error: 

```
>>> import torch
>>> a = torch.rand(3)
>>> a.expand(-1, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: The expanded size of the tensor (-1) isn't allowed in a leading, non-existing dimension 0
```

while your shape inference would silently estimate `target_size[0]` as `0`?